web/src/components/Header/HeaderRightPanel.tsx

```diff
@@ -64,96 +100,180 @@ function HeaderRightPanel({
   const searchParams = new URLSearchParams(location.search)
   const isGuideOpen = !!searchParams.get(GUIDE_IS_OPEN)
 
+  const searchResults = []
+
   return (
-    <div className="header-right-panel">
-      {/* <Icon name="search-line.svg" onClick={clickSearch}/> */}
-      {/* open or close the guidebook, depending on if it */}
-      {/* is currently open or closed */}
-      {/* Guidebook Button */}
-      {/* @ts-ignore */}
-      <NavLink
-        className="header-right-panel-icon"
-        to={`${location.pathname}${
-          isGuideOpen ? '' : '?' + GUIDE_IS_OPEN + '=1'
-        }`}
-        // if clicked on guidebook for the first time, remove the help message
-        // and remember not to show that in the future (store that locally) aka persist
-        onClick={hideGuidebookHelpMessage}
+    <>
+      <div
+        className={`search-button-wrapper ${
+          isSearchOpen ? 'search-is-open' : ''
+        } 
+        ${filterText !== '' ? 'results-dropdown-is-open' : ''}`}
       >
+        <div className="search-icon-input">
+          <div className="search-open-icon">
+            <Icon
+              name="search.svg"
+              size="small"
+              onClick={() => {
+                setIsSearchOpen(!isSearchOpen)
+                setFilterText('')
+              }}
+            />
+          </div>
+          <div>
+            <CSSTransition
+              in={isSearchOpen}
+              timeout={100}
+              unmountOnExit
+              classNames="search-input-wrapper"
+            >
+              <input
+                type="text"
+                onChange={(e) => setFilterText(e.target.value.toLowerCase())}
+                value={filterText}
+                placeholder="Search for a goal, comment, and more"
+                autoFocus
+              />
+            </CSSTransition>
+          </div>
+          {filterText !== '' && (
+            <Icon
+              name="x.svg"
+              size="small"
+              className="light-grey"
+              onClick={() => {
+                setFilterText('')
+              }}
+            />
+          )}
+        </div>
+        {/* <CSSTransition
+          in={filterText !== ''}
+          timeout={200}
+          unmountOnExit
+          className="search-results-dropdown-wrapper"
+        > */}
+        {filterText !== '' && (
+          <div className="search-results-dropdown">
+            <div className="search-results-filters">
+              <SearchResultsFilter />
+              <SearchResultsFilter />
+              <SearchResultsFilter />
+            </div>
+            <div className="search-results-list">
+              {goals.filter((goal) => (
+                  goal["content"].includes(filterText)
```

Comment:
```suggestion
                  goal.content.includes(filterText)
```

both ways to access a value work, this is the preferred way though